### ISSUE TITLE
URLEncode with unicode characters is malformed inside a Querystring

### ISSUE THREAD

**Issue by arnoldpistorius** [canton7/RestEase#46]
Hi,

I found a bug in the Query building part of RestEase.

In `Requester.cs` line 132 the following code does a double encode, which causes the character `é` to become `%u00e9` first (from the `QueryParamBuilder`), and finally `%25u00e9` when the `uriBuilder.Uri` is returned.

```
uriBuilder.Query = QueryParamBuilder.Build(initialQueryString, queryParams).TrimStart('?');

return uriBuilder.Uri;
```

In other words the `QueryParamBuilder` class does an encode, and the `UriBuilder` class does another encode. So the string get encoded twice.

**Comment by canton7** [comment]
Interesting, thanks for reporting.

**Comment by canton7** [comment]
Note: Only an issue on .NET framework (not .NET Core)

**Comment by canton7** [comment]
In fact, it's the HttpValueCollection that's double-encoding that unicode character.

**Comment by canton7** [comment]
This actually has a good side-effect, in that I've found a way to encode query parameters which doesn't result in platform-specific code (and in copy-pasting a bunch of third-party libraries for .NET Core).

**Comment by canton7** [comment]
Leaving open for visibility.

**Comment by arnoldpistorius** [comment]
@canton7 you're the greatest!

Any idea when the next release will hit nuget repo?

**Comment by canton7** [comment]
I messed up and left some work I wanted to include in this release on my computer at home, so maybe tonight, maybe sometime over the weekend.

**Comment by canton7** [comment]
You can grab intermediate builds off AppVeyor if you need a short-term fix. https://ci.appveyor.com/nuget/restease

**Comment by canton7** [comment]
Released in 1.4.1
